plutus-core/plutus-core/src/Data/SatInt.hs

```diff
@@ -246,21 +265,52 @@ quotRemSI a@(I# _) b@(I# _) = (SI (a `quotInt` b), SI (a `remInt` b))
 divModSI ::  Int -> Int -> (SatInt, SatInt)
 divModSI x@(I# _) y@(I# _) = (SI (x `divInt` y), SI (x `modInt` y))
 
+
 plusSI :: SatInt -> SatInt -> SatInt
+plusSI (SI (I# x#)) (SI (I# y#)) =
+  case addIntC# x# y#  of
+    (# r#, f# #) ->
+        if isTrue# f# then  -- Overflow;  I think the value of f# is always 1 in this case
+                            -- and doesn't give us any other useful information
+        if      isTrue# ((x# ># 0#) `andI#` (y# ># 0#)) then maxBound
+        else if isTrue# ((x# <# 0#) `andI#` (y# <# 0#)) then minBound
+        else 0  -- Shouldn't happen: error?  `succ maxBound` already raises an exception.
```

Comment:
This is when x and y have opposite signs, right? I agree, I don't think that should be able to overflow.

Reply:
Yes, that was my thinking.